quantum/quantum.h

```diff
@@ -197,6 +197,8 @@ typedef uint8_t pin_t;
 
 #    define readPin(pin) ((bool)(PINx_ADDRESS(pin) & _BV((pin)&0xF)))
 
+#    define togglePin(pin)  (PINx_ADDRESS(pin) |= _BV((pin)&0xF))
```

Comment:
Shouldn't this be:
```suggestion
#    define togglePin(pin)  (PORTx_ADDRESS(pin) ^= _BV((pin)&0xF))
```

Reply:
Take a look at the AVR datasheet.
Surprisingly, writing bit one to the PIN register has the ability to invert the corresponding bit of the PORT register.

Reply:
Huh, that's handy.

Reply:
Actually... looking at the 32A datasheet this seems to not always be the case, at least, I don't see the same text in that section. Could just be a mistake though. AT90USB64/128, 32U2 and 328P all seem to be fine.

Reply:
I'll write a test program to confirm.

Reply:
I wrote a test program on ProMicro and Arduino UNO to check the GPIO output toggle.
https://github.com/mtei/AVR_GPIO_toggle_test

Reply:
Alternatively, you can use the PIN register toggle function with the following conditions.

```suggestion
#    if defined(__OPTIMIZE__) && (defined(__AVR_ATmega32U4__) || defined(__AVR_ATmega328P__))
#        define togglePin(pin)  (PINx_ADDRESS(pin) |= _BV((pin)&0xF))
#    else
#        define togglePin(pin)  (PORTx_ADDRESS(pin) ^= _BV((pin)&0xF))
#    endif
```

Reply:
@mtei that suggestion misses a lot of the supported AVR chips. It seems only the 32A does not have this functionality (even the ATtiny85 does!).

Tested on JJ4x4 with:
```c
bool process_record_user(uint16_t keycode, keyrecord_t *record) {
    switch (keycode) {
        case MY_TOGG:
            if (record->event.pressed) {
                setPinOutput(D4);
                PIND |= (1 << 4);
            }
            return false;
    }
    return true;
}
```
When I change it to `PORTD ^= (1 << 4)` the backlight toggles as expected.
So, my initial suggestion is better as we apparently can't rely on this behaviour.